clouds/aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,793 @@
+        subnet = ec2_client.create_subnet(
+            CidrBlock=subnet_cidr_blocks[count],
+            VpcId=vpc_id,
+            AvailabilityZoneId=availability_zones[count],
+        )
+        subnet_details = ec2_client.describe_subnets(
+            SubnetIds=[subnet["Subnet"]["SubnetId"]]
+        )
+        subnet_details = ec2_client.describe_subnets(
+            SubnetIds=[subnet["Subnet"]["SubnetId"]]
+        )
+        try:
+            for sample in TimeoutSampler(
+                wait_timeout=300,
+                sleep=5,
+                func=is_resource_available,
+                aws_resource={"Resource": subnet_details["Subnets"][0]},
+            ):
+                if sample is True:
+                    break
+        except TimeoutExpiredError:
+            LOGGER.error(f" Subnet {subnet['Subnet']['SubnetId']} not available.")
+            raise
+
+        ec2_assign_name_tag(
+            ec2_client=ec2_client,
+            res_id=subnet["Subnet"]["SubnetId"],
+            tag_name=f"{site_name}-subnet-public-{count+1}",
+        )
+        public_subnet_ids_list.append(subnet["Subnet"]["SubnetId"])
+    return public_subnet_ids_list
+
+
+def create_private_subnets_for_vpc(
+    ec2_client, subnet_count, vpc_id, subnet_cidr_blocks, availability_zones, site_name
+) -> list:
+    """
+    Creates public subnets in a given vpc
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         subnet_count (int): The number of additional bits with which to extend the prefix
+         vpc_id (str): Id for the VPC where subnets need to be created
+         subnet_cidr_blocks (list): list of subnets CIDRs available
+         availability_zones (list): list of available zones
+         site_name (str): The value for 'Name' tag used for all resource
+
+    Return:
+         list: list of ID's for the subnets created
+    """
+    LOGGER.info(f"Creating {subnet_count} Private Subnets.")
+    private_subnet_ids_list = []
+    for count in range(subnet_count):
+        subnet = ec2_client.create_subnet(
+            CidrBlock=subnet_cidr_blocks[count + subnet_count],
+            VpcId=vpc_id,
+            AvailabilityZoneId=availability_zones[count],
+        )
+        subnet_details = ec2_client.describe_subnets(
+            SubnetIds=[subnet["Subnet"]["SubnetId"]]
+        )
+        try:
+            for sample in TimeoutSampler(
+                wait_timeout=300,
+                sleep=5,
+                func=is_resource_available,
+                aws_resource={"Resource": subnet_details["Subnets"][0]},
+            ):
+                if sample is True:
+                    break
+        except TimeoutExpiredError:
+            LOGGER.error(f"Subnet {subnet['Subnet']['SubnetId']} not available.")
+            raise
+
+        ec2_assign_name_tag(
+            ec2_client=ec2_client,
+            res_id=subnet["Subnet"]["SubnetId"],
+            tag_name=f"{site_name}-subnet-private-{count+1}",
```

Comment:
Is the only different between this function and `create_public_subnets_for_vpc` is 
`tag_name=f"{site_name}-subnet-private-{count+1}",` ?

If so please extract the common code to function